app/questionnaire/routing/operations.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+from typing import Iterable, Optional, Sequence, Union
+
+from dateutil.relativedelta import relativedelta
+
+from app.questionnaire.routing.helpers import casefold, datetime_as_midnight
+from app.questionnaire.rules import convert_to_datetime
+
+answer_types = Union[bool, str, int, float, None, datetime]
+
+
+def _are_non_none_operands(*operands: Union[Sequence, answer_types]) -> bool:
+    return all(operand is not None for operand in operands)
+
+
+@casefold
+def evaluate_equal(lhs: answer_types, rhs: answer_types) -> bool:
+    return lhs == rhs
+
+
+@casefold
+def evaluate_not_equal(lhs: answer_types, rhs: answer_types) -> bool:
+    return lhs != rhs
+
+
+def evaluate_greater_than(lhs: answer_types, rhs: answer_types) -> bool:
+    return _are_non_none_operands(lhs, rhs) and lhs > rhs  # type: ignore
```

Comment:
Ah nice one didn't think of that. So this gets us mostly there, but I would have to check that the operands are not None inline for each operation for it to work. As it can't work out we are check for None in `_are_non_none_operands`. So:
```py
def evaluate_greater_than(lhs: AnyValue, rhs: AnyValue) -> bool:
    return lhs is not None and rhs is not None and lhs > rhs
```

Thoughts?

Reply:
Another option could be doing the check before the method is called? (for this we would need to create a list of operations that don't work with `None` values).

Reply:
I did have that initially but then thought each operation should be self-contained, but will revisit given it makes typing better.

Reply:
Operations that require non NoneType values are now explicitly checked. That, with the use of `TypeVar` the mypy ignore comment have been removed.